site/src/Controller/NotificationController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Notification;
+use Mgilet\NotificationBundle\NotifiableInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Mgilet\NotificationBundle\Manager\NotificationManager;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class NotificationController
+ */
+class NotificationController extends AbstractController
+{
+    /**
+     * @var NotificationManager
+     */
+    protected $notificationManager;
+
+    /**
+     * NotificationController constructor.
+     *
+     * @param NotificationManager $notificationManager
+     */
+    public function __construct(NotificationManager $notificationManager)
+    {
+        $this->notificationManager = $notificationManager;
+    }
+
+    /**
+     * Mark a Notification as seen
+     *
+     * @Route("/notification/seen", name="notification_mark_seen")
+     *
+     * @Method("POST")
+     *
+     * @param Request $request
+     *
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws \Doctrine\ORM\EntityNotFoundException
+     * @throws \Exception
+     *
+     * @return JsonResponse
+     */
+    public function markAsSeenAction(Request $request)
+    {
+        $notification = $request->get('notification');
+        $notifiable = $request->get('notifiable');
```

Comment:
@imphil Now the logic is moved to userController where we fetch the user in the controller itself rather than sending it with the request. We get the notifiable accordingly there. Please have a check is the current approach alright. I have tested and it seems to work well